shapeshift/web
# feat: warn against high price impact at confirm step

## Description

~~Now that we have custom slippage under flag, we can implement https://github.com/shapeshift/web/issues/4858, since users may input custom slippage up to 30%, effectively having a maximum downside of 30% on the trade itself (excluding fees, which are separate from the trade value and are typically paid in addition to it for most swappers).~~

Disregard this. This is only implemented for price impact (the fiat difference between what you're selling and what you're getting), discusses this one out with @reallybeard and confirmed having slippage under the same umbrella would be all sorts of weird.

## Pull Request Type


## Issue (if applicable)



closes https://github.com/shapeshift/web/issues/4858

## Risk



This may show when it shouldn't, test accordingly

## Testing



- Try an unfavorable trade e.g UTXO to ETH with THOR and the smallest amount you can swap (e.g 100 DOGE to ETH, might be more like 200 at the time you're testing this)
- Note, we don't properly handle minimums and balance checks after the multi-hop refactor (https://github.com/shapeshift/web/issues/5011) so you'll see weird negative values until we get that fixed, if you're under the minimum or the quote fails for any reason related to your balance
- Ensure the price impact is reflected with a confirm window
- Disregard me ~~With the custom slippage flag on, add custom slippage e.g 9% (which is lower than the 10% minimum heuristics to display the confirm window), and ensure that the unfavorable downside (~60% at the time of testing) takes precedence over the custom slippage~~
- Disregard me ~~Repeat the above with a favorable trade (e.g ERC-20 -> ERC-20 on Gnosis) and high custom slippage, ensuring the high custom slippage is reflected with a confirmed window, not the price impact downside (which should be very low, closeish to 0)~~

### Engineering



### Operations



## Screenshots (if applicable)

- High price impact

<img width="654" alt="image" src="https://github.com/shapeshift/web/assets/17035424/9a809970-1d06-412c-b556-96ba1971a45b">


- Disregard me ~~High slippage taking precedence over low price impact~~

<img width="641" alt="image" src="https://github.com/shapeshift/web/assets/17035424/640c74f2-e501-4cc7-9e55-fad492cab1bb">

- Disregard me ~~High price impact taking precedence over low slippage~~

<img width="629" alt="image" src="https://github.com/shapeshift/web/assets/17035424/345b2747-fa32-4fd0-8ace-76b1a4b59a56">

- Disregard me~~High slippage~~

<img width="618" alt="Screenshot 2023-08-04 at 13 52 44" src="https://github.com/shapeshift/web/assets/17035424/ccaafd4c-af21-4ba3-a4fd-26c432746bab">
